openerp/models.py

```diff
@@ -4499,9 +4498,18 @@ def _generate_order_by(self, order_spec, query):
                             order_by_elements.append("%s %s" % (clause, order_direction))
                     else:
                         order_by_elements.append("%s %s" % (inner_clause, order_direction))
-            if order_by_elements:
-                order_by_clause = ",".join(order_by_elements)
+            return order_by_elements
+        return []
 
+    def _generate_order_by(self, order_spec, query):
+        """
+        Attempt to consruct an appropriate ORDER BY clause based on order_spec, which must be
+        a comma-separated list of valid field names, optionally followed by an ASC or DESC direction.
+
+        :raise" except_orm in case order_spec is malformed
```

Comment:
I guess a `:` is missing after `raise`  I think correct usage is `:raises ExceptionType: docstring`